datadog_checks_dev/datadog_checks/dev/tooling/commands/dep.py

```diff
@@ -108,3 +117,175 @@ def freeze():
+            for check_dependency_name, check_dependency_definitions in check_dependencies.items()
+            for version, dependency_definitions in check_dependency_definitions.items()
+            for check_dependency_definition in dependency_definitions
+            for agent_dep_version, agent_dependency_definitions in all_agent_dependencies[check_dependency_name].items()
+            for agent_dependency_definition in agent_dependency_definitions
+            # look for the dependency with the same marker and name since the version can be different
+            if check_dependency_definition.same_name_marker(agent_dependency_definition)
+            and version != agent_dep_version
+        }
+
+        if deps_to_update:
+            files_updated += 1
+            check_req_file = get_check_req_file(check_name)
+            old_lines = read_file_lines(check_req_file)
+            new_lines = old_lines.copy()
+
+            for dependency_definition, new_version in deps_to_update.items():
+                dependency_definition.requirement.specifier = new_version
+                new_lines[dependency_definition.line_number] = f'{dependency_definition.requirement}\n'
+
+            write_file_lines(check_req_file, new_lines)
+
+    if not files_updated:
+        echo_info('All dependencies synced.')
+
+    echo_info(f'Files updated: {files_updated}')
+
+
+async def get_version_data(url):
+    async with request('GET', url) as response:
+        try:
+            info = orjson.loads(await response.read())['info']
+        except Exception as e:
+            raise type(e)(f'Error processing URL {url}: {e}')
+        else:
+            return (
+                info['name'],
+                info['version'],
+                {
+                    extract_classifier_value(c)
+                    for c in info['classifiers']
+                    if c.startswith('Programming Language :: Python ::')
+                },
+            )
+
+
+async def scrape_version_data(urls):
+    package_data = defaultdict(lambda: {'version': '', 'classifiers': set()})
+
+    async with Pool() as pool:
+        async for package_name, package_version, version_classifiers in pool.map(get_version_data, urls):
+            data = package_data[package_name]
+            if package_version:
+                data['version'] = package_version
+
+            data['classifiers'].update(version_classifiers)
+
+    return package_data
+
+
+def is_version_compatible(marker, supported_versions):
+    """
+    Determines if any of the given versions are compatible with the given marker
+    """
+    os_markers = ['darwin', 'win32']
+    is_compatible = False
+
+    if marker is None:
+        # manually set marker since if there's none the dependency should work on all environments
+        marker = Marker(
+            'sys_platform == "win32" or sys_platform == "darwin" and python_version < "3.0" or python_version > "3.0"'
```

Comment:
This doesn't match everything, no?

Reply:
Maybe simply `python_version >= "2.0"`

Reply:
Should these still be added to https://github.com/DataDog/integrations-core/pull/9811/files#diff-2a97859c44f7b3be66b14848ac7c1e0dd461ae8947803923cb4ddb2df6475617R210?

Reply:
yes except cygwin

Reply:
Thanks, updated!